Tensile/TensileRetuneLibrary.py

```diff
@@ -0,0 +1,173 @@
+    # could use a refactor to elimate duplicated code
+    ClientExecutable.getClientExecutable()
+
+    shortName = "benchmark"
+    benchmarkDir = os.path.join(outPath, shortName)
+    sourceDir = os.path.join(benchmarkDir, "source")
+    ensurePath(sourceDir)
+
+    pushWorkingPath(shortName)
+    pushWorkingPath("source")
+
+    filesToCopy = [
+        "SolutionMapper.h",
+        "Client.cpp",
+        "Client.h",
+        "CMakeLists.txt",
+        "DeviceStats.h",
+        "TensorUtils.h",
+        "MathTemplates.cpp",
+        "MathTemplates.h",
+        "TensileTypes.h",
+        "tensile_bfloat16.h",
+        "KernelHeader.h",
+        "ReferenceCPU.h",
+        "SolutionHelper.cpp",
+        "SolutionHelper.h",
+        "Tools.cpp",
+        "Tools.h",
+        ]
+
+    for f in filesToCopy:
+        shutil.copy(
+            os.path.join(globalParameters["SourcePath"], f),
+            globalParameters["WorkingPath"] )
+    if globalParameters["RuntimeLanguage"] == "OCL":
+        shutil.copy(
+            os.path.join(globalParameters["SourcePath"], "FindOpenCL.cmake"),
+            globalParameters["WorkingPath"] )
+    else:
+        shutil.copy(
+            os.path.join(globalParameters["SourcePath"], "FindHIP.cmake"),
+            globalParameters["WorkingPath"] )
+
+    # make directory for results and set update yaml file
+    resultsDir = os.path.normpath(os.path.join(globalParameters["WorkingPath"], "../../Data"))
+    ensurePath(resultsDir)
+    updateFile = os.path.join(resultsDir, "update.yaml")
+    Common.globalParameters["LibraryUpdateFile"] = updateFile
+
+    BenchmarkProblems.writeBenchmarkFiles(benchmarkDir, solutions, problemSizes, shortName, filesToCopy, [])
+
+    popWorkingPath() # source
+
+    libraryLogicPath = None
+    forBenchmark = True
+    # TODO make this work with TileAware selection
+    returncode = ClientWriter.runClient(libraryLogicPath, forBenchmark, False)
+
+    if returncode:
+        printExit("BenchmarkProblems: Benchmark Process exited with code %u" % returncode)
+
+    return updateFile
+
+
+def TensileRetuneLibrary(userArgs):
+    print1("")
+    print1(HR)
+    print1("#")
+    print1("#  Tensile Retune Library v{}".format(__version__))
+
+    # setup argument parsing
+    argParser = argparse.ArgumentParser()
+    argParser.add_argument("library_file", type=os.path.realpath, help="library logic file to retune")
+    argParser.add_argument("output_path", help="path where to conduct benchmark")
+    addCommonArguments(argParser)
+    args = argParser.parse_args(userArgs)
+
+    libPath = args.library_file
+
+    print1("#  Library Logic: {}".format(libPath))
+    print1("#")
+    print1(HR)
+    print1("")
+
+    # setup global parameters
+    outPath = ensurePath(os.path.abspath(args.output_path))
+    restoreDefaultGlobalParameters()
+    assignGlobalParameters({"LibraryFormat": "msgpack",
+                            "OutputPath": outPath,
```

Comment:
I am assuming this narrow case only considers msgpack output? Or would we also need to consider yaml?

Reply:
This just controls the format for the master solution library that will be used by the Tensile client, which is just an intermediary for producing the final result. The overall input and output for this tool is a library logic file, which is always yaml. 

Using the yaml library format should work just fine, but I simply chose to hardcode for msgpack due to the narrow intended use case of this tool.